shumai/tensor/tensor.ts

```diff
@@ -280,6 +280,11 @@ export class Tensor {
     )
   }
 
+  private track_tidy_ptr(ptr?: number) {
+    if (!ptr) ptr = this.ptr
+    if (!_tidyTracker.has(ptr)) _tidyTracker.set(ptr, this)
```

Comment:
why would _tidyTracker ever have the pointer already in it?

Reply:
circular references